### PR TITLE
[BLOCKED] Removed MHV terms and conditions requirement for account creation

### PR DESCRIPTION
**Pull request by U-DON**
## Description of change
MHV account state should no longer go through the `needs_terms_acceptance` state as it's no longer required for account creation. T&C status is no longer being serialized for `GET /mhv_account`. T&C data is no longer passed in the payload to MHV for the account creation request.

FE PR: department-of-veterans-affairs/vets-website#9110

## Testing done
Local testing on user vets.gov.user+1 comparing what happens with and without changes on both BE and FE. Without the changes, clicking the button to create an account will redirect to T&C (due to the account state being `needs_terms_acceptance`). With the changes, clicking it will start a `POST /mhv_account` request (due to the account state being `no_account`).

## Testing planned
Should test the account upgrade flow on Staging. Perhaps after removing (unless we decide not to) the VA patient requirement as well.

## Acceptance Criteria (Definition of Done)
- [x] Users eligible for MHV account creation should not have to accept T&C in order to proceed with account creation.

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by saneshark** [comment]
@U-DON keep in mind, we should verify on production that Needs Terms is not a terminal state that the state machine could have ended up in. Otherwise, we need to at least handle the transition from that state to another state.

**Comment by U-DON** [comment]
I think it's unlikely that there are accounts in a terminal state of `needs_terms_acceptance`. But even if there are, they should still proceed to `eligible` and so forth since the initial transitions in the `check_eligibility` step occur for `ALL_STATES`.

File: spec/models/mhv_account_spec.rb [hunk 1]
```diff
@@ -164,170 +156,150 @@
...
-        before(:each) { create(:terms_and_conditions_acceptance, terms_and_conditions: terms, user_uuid: user.uuid) }
-
-        context 'without an existing account' do
-          context 'nothing has been persisted and no mhv_id' do
-            it 'has no_account' do
-              expect(subject.account_state).to eq('no_account')
-              expect(subject.creatable?).to be_truthy
-              expect(subject.terms_and_conditions_accepted?).to be_truthy
-            end
+      context 'without an existing account' do
```
Comment (U-DON) [inline spec/models/mhv_account_spec.rb]: This whole section was just moving everything out of the `with terms accepted` context block.

File: spec/request/mhv_accounts_request_spec.rb [hunk 2]
```diff
@@ -147,146 +143,134 @@
...
-                                             message: V0::MhvAccountsController::CREATE_ERROR
-    it_behaves_like 'a failed POST #upgrade', http_status: :forbidden,
-                                              message: V0::MhvAccountsController::UPGRADE_ERROR
-    it_behaves_like 'ssn mismatch'
-    it_behaves_like 'non va patient'
-  end
-
-  context 'with T&C acceptance' do
-    before { create(:terms_and_conditions_acceptance, terms_and_conditions: terms, user_uuid: user.uuid) }
+  it_behaves_like 'ssn mismatch'
```
Comment (U-DON) [inline spec/request/mhv_accounts_request_spec.rb]: Similarly, all of the following changes are just shifting blocks out of the T&C contexts.

File: lib/mhv_ac/upgrade_form.rb [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
-require 'common/models/attribute_types/httpdate'
 
 module MHVAC
   class UpgradeForm < Common::Form
     include ActiveModel::Validations
 
     attribute :user_id, Integer
-    attribute :form_signed_date_time, Common::HTTPDate
-    attribute :form_upgrade_manual_date, Common::HTTPDate
-    attribute :form_upgrade_online_date, Common::HTTPDate
```
Comment (saneshark) [inline lib/mhv_ac/upgrade_form.rb]: Are we sure that the MHV api no longer requires us to provide these?

File: spec/lib/mhv_ac/client_spec.rb [hunk 4]
```diff
@@ -36,18 +36,14 @@
...
+      sign_in_partners: 'VA.GOV',
+      email: 'vets.gov.user+0@gmail.com'
     }
   end
 
   let(:upgrade_params) do
     {
-      user_id: '14221465',
-      form_signed_date_time: time.httpdate,
-      terms_version: 'v3.4'
```
Comment (saneshark) [inline spec/lib/mhv_ac/client_spec.rb]: Unless MHV has made changes to their API, I would think these are still required.

File: lib/mhv_ac/upgrade_form.rb (hunk 3 above)
Comment (U-DON) [reply lib/mhv_ac/upgrade_form.rb]: I think they may still requirement. From a local test, I'm getting a "data integrity issue" from the upgrade request. Should we just use the current date or work with MHV to remove the requirement?

Comment (U-DON) [reply lib/mhv_ac/upgrade_form.rb]: I've also gotten an "unknown application error" from trying to upgrade another account (Advanced).

**Comment by annaswims** [comment]
Do we have documentation on the MHV API somewhere?

File: lib/mhv_ac/upgrade_form.rb (hunk 3 above)
Comment (saneshark) [reply lib/mhv_ac/upgrade_form.rb]: They should remove the requirement. It doesn't make sense for us to say the user signed a form or accepted T&C if they didn't actually do so.

Comment (saneshark) [reply lib/mhv_ac/upgrade_form.rb]: @davidsbao thoughts?

**Comment by U-DON** [comment]
This is blocked until we discuss with MHV about changes to their account creation APIs. We no longer need to pass the T&C data and should coordinate to remove the requirement (if there is one). Since we are going to be providing other data in the SSO cookie, we may eventually get to a point where the account upgrades can happen on the MHV side entirely.

**Comment by BBarone1** [comment]
@U-DON and @saneshark - we've decided to keep T&C for the near term on va.gov and no longer need this work at this time. OK to close this PR?

**Comment by U-DON** [comment]
@BBarone1: It should be okay to close. Are we expecting to remove the T&C longer term, though?

**Comment by U-DON** [comment]
We've decided to keep T&C, so closing.

**Pull request closed without merging**